client/my-sites/checkout/index.js

```diff
@@ -51,13 +51,7 @@ export default function () {
 
 	// Handle logged-in user visiting Jetpack checkout
 	if ( isEnabled( 'jetpack/userless-checkout' ) ) {
-		page(
-			'/checkout/jetpack/:product/:domainOrProduct',
-			siteSelection,
-			checkout,
-			makeLayout,
-			clientRender
-		);
```

Comment:
Hey @sirbrillig, do you know if this route is being used? I understand that `/checkout/jetpack` was added recently to make the logged-out checkout work, but I don't understand why a route that enables domain purchases was added. Are Jetpack sites supposed to purchase domains?

Reply:
> It might be because checkout (without jetpack anyway) accepts URLs of either /checkout/:site/:product or /checkout/:product/:site (a source of some confusion that I'd love to get rid of) so perhaps :product here means "product or site" and :domainOrProduct means "site or product"?

That makes sense, and I agree that we shouldn't have two routes that lead to the same place.

@rcoll, do you happen to recall why that route was added?